pages/api/auth/[...nextauth].ts

```diff
@@ -0,0 +1,99 @@
+import NextAuth from "next-auth"
+import { JWT } from "next-auth/jwt"
+import ADProvider from "next-auth/providers/azure-ad"
+import qs from "query-string"
+
+import { config } from "../../../config"
+import { HttpClient } from "../../../lib/HttpClient"
+
+const SCOPE = "openid offline_access https://equinor-dev.collibra.com/user_impersonation"
+
+type Token = JWT & {
+  tokenType: string,
+  accessToken: string,
+  expiresAt: number,
+  refreshToken: string,
+  error?: string
+}
+
+type RefreshTokenResponse = {
+  token_type: string
+  scope: string
+  expires_in: number,
+  ext_expires_in: number,
+  access_token: string,
+  refresh_token: string,
+  id_token: string
+}
+
+const tryAccessTokenRefresh = async (token: Token): Promise<Token> => {
+  try {
+    const res = await HttpClient.post<RefreshTokenResponse>(`https://login.microsoftonline.com/${config.AUTH_TENANT_ID}/oauth2/v2.0/token`, {
```

Comment:
Question: Should we avoid hard coded urls like this and rather add them to some config? 🤔

Reply:
my take on this is that if it's needed multiple places in the code, then yes. what's your take here?

Reply:
Jeg er pragmatisk her, men reagerer instinktivt når jeg ser url-er som tekst i kode bare 😅  Syns det er helt OK og ha det slik så lenge